File: package/website/src/api/ocr.ts
import request from '@/utils/request';

export interface OCRRecord {
  id: number
  photo_id: string
  text: string
  text_score: number
  polygon: number[][] // [[x,y], [x,y], ...]
}

export interface OCRResponse {
  count: number
  records: OCRRecord[]
}

export const ocrApi = {
  async getOCR(photoId: string) {
    const data = await request.get<OCRResponse>('/api/ocr', { 
        params: { photo_id: photoId } 
    })
    return data
  },

  async deleteOCR(photoId: string) {
    const data = await request.delete(`/api/ocr/${photoId}`)
    return data
  }
}

File: package/server/app/schemas/ocr.py
from typing import List, Optional
from pydantic import BaseModel
from uuid import UUID

class OCRBase(BaseModel):
    text: str
    text_score: float
    polygon: List[List[float]] # [[x1, y1], [x2, y2], ...] normalized 0-1

class OCRCreate(OCRBase):
    photo_id: UUID

class OCR(OCRBase):
    id: int
    photo_id: UUID

    class Config:
        from_attributes = True

class OCRResponse(BaseModel):
    count: int
    records: List[OCR]
